shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs

```diff
@@ -373,15 +362,15 @@ instance
   CC.Crypto crypto =>
   FromCBOR (DState crypto)
   where
-  fromCBOR = do
-    decodeRecordNamed "DState" (const 6) $ do
-      rw <- fromCBOR
-      dlg <- fromCBOR
-      p <- fromCBOR
-      fgs <- fromCBOR
-      gs <- fromCBOR
-      ir <- fromCBOR
-      pure $ DState rw dlg p fgs gs ir
+  fromCBOR =
+    decodeRecordNamed "DState" (const 6) $
+      DState
+        <$> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
+        <*> fromCBOR
```

Comment:
I think the named version is slightly easier to read here

Reply:
Reverted